src/cdk-experimental/dialog/dialog.ts

```diff
@@ -67,8 +71,8 @@ export class Dialog {
   constructor(
       private overlay: Overlay,
       private injector: Injector,
-      @Inject(DIALOG_REF) private dialogRefConstructor,
-      @Inject(DIALOG_SCROLL_STRATEGY) private _scrollStrategy,
+      @Inject(DIALOG_REF) private dialogRefConstructor: Type<DialogRef<any>>,
+      @Inject(DIALOG_SCROLL_STRATEGY) scrollStrategy: any,
```

Comment:
Why can't this be `ScrollStrategy`?

(same for other places that inject a strategy)

Reply:
It's actually a `() => ScrollStrategy`. It can't have the actual type in here, because the compiler complains about it. I tried introducing a new type called `ScrollStrategyFactory`, but that ended up breaking in a different way.

Reply:
Could you add a TODO to change it back once we're using the ivy renderer? It will fix the problem that prevents the correct type from being used.

Reply:
Done.